### ISSUE TITLE
occasionally fails with "ERROR: relation "okta_user" does not exist (SQLSTATE 42P01)"

### ISSUE THREAD

**Issue by foo-1a** [turbot/steampipe-plugin-okta#85]
**Describe the bug**
I am running regular reports with Steampipe from a scheduleg Github Action.
I have a custom docker action in which the `Dockerfile`  pre-installs the this Okta plugin.
The action entrypoint then does a `steampipe service start` then `steampipe query "<query-as-action-arg>"`.

This sometimes work just fine, and sometimes (roughly 50% of the time actually) fails, this is the output that I get in GHA's log:
```
Steampipe service is running:

Database:

  Host(s):            localhost, 127.0.0.1, 172.17.0.2
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show*** reveal]
  Connection string:  postgres://steampipe@localhost:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status
	 
  # View database password for connecting from another machine
  steampipe service status --show-password
  
  # Restart the service
  steampipe service restart
  
  # Stop the service
  steampipe service stop

Warning: executeQueries: query 1 of 1 failed: ERROR: relation "okta_user" does not exist (SQLSTATE 42P01)
```

Any idea of what might this be, or pointers on how I can further debug this?

**Steampipe version (`steampipe -v`)**
v0.20.9

**Plugin version (`steampipe plugin list`)**
v0.9.0

**To reproduce**
I can provide a more detailed setup for the action if needed for further analysis.

**Expected behavior**
Expected the `okta_user` table to exist given that the plugin is pre-installed during the action's docker build

**Additional context**
n/a

**Comment by binaek** [comment]
@foo-1a thank you for reporting this.

It looks like there may be a problem with the retrying code logic which waits for the connections to load when it sees a `relation not found` error from the service.

Can you update the workflow and add a ::
`steampipe query "select name,state,import_schema,plugin,connection_mod_time,plugin_mod_time from steampipe_connection_state"` right after your `steampipe query`.

The `steampipe_connection_state` table contains the state information on all the connections that `steampipe` loads. Ideally, they should all be in the `ready` state.

**Comment by foo-1a** [comment]
I updated the action to also run that query as a second command - oddly enough I get this output in a run that actually succeeds:
```
Error: Reattachment process not found
+-----------+---------+---------------+--------------------------------------------------+----------------------+----------------------+
| name      | state   | import_schema | plugin                                           | connection_mod_time  | plugin_mod_time      |
+-----------+---------+---------------+--------------------------------------------------+----------------------+----------------------+
| steampipe | pending | enabled       | hub.steampipe.io/plugins/turbot/steampipe@latest | 2023-08-09T08:04:56Z | 2023-08-09T08:04:38Z |
| okta      | pending | enabled       | hub.steampipe.io/plugins/turbot/okta@latest      | 2023-08-09T08:04:56Z | 2023-08-09T08:04:41Z |
+-----------+---------+---------------+--------------------------------------------------+----------------------+----------------------+
```
have not observed a failing run since adding this yesterday - will update here once I do.
version `v0.9.0`

**Comment by pskrbasu** [comment]
@foo-1a We released v0.20.10 of Steampipe with a potential fix for this issue. Please take a look if you still see the `relation does not exist error` in your workflow.

Thanks

**Comment by foo-1a** [comment]
I've not seen errors popping up anymore since Aug.
This can be closed, thanks!
